src/main/java/teammates/common/util/StringHelper.java

```diff
@@ -279,7 +281,7 @@ public static String removeExtraSpace(String str) {
         if (str == null) {
             return null;
         }
-        return str.trim().replaceAll("\\s+", " ");
+        return CharMatcher.whitespace().trimFrom(str).replaceAll("\\s+", " ");
```

Comment:
I'm just curious about the collapsing behavior. We are trimming all kinds of whitespaces, but only collapsing "regex-type" whitespaces, is that intended?

Reply:
Yes; otherwise we are changing existing behaviour.

Reply:
Feels like strange behavior to keep non-breaking spaces to me. And keeping multiple non-breaking spaces in between seems even weirder?

Reply:
What if the user really wants a non-breaking space as part of their name, for e.g. to ensure that their name isn't broken across two lines even though it's two words long?

Reply:
But I don't think the formatting should be mixed together with the data?

Reply:
But it's a valid utf-8 character that we support?

Reply:
But aren't we similarly supporting other whitespaces? Why are we collapsing the other whitespaces but not collapsing non-breaking spaces? If we want to preserve multiple non-breaking spaces, that seems to be a (slight) deviation from non-breaking space behavior?

Reply:
This discussion is already out of the scope of "**leading spaces** in email should be trimmed internally". Let's not stall the PR for that. On a more general note, let's not stall any PR for issues not related to the work done, unless they are critical.

Reply:
The reason I had a question is because it is not clear from the code what the purpose is. If anything if we do a `trimAndCollapseFrom` the behavior might change but simpler and easier to reason about.

And if we do trim all leading spaces and not collapse all spaces, it creates a kind of inconsistency, so I don't think this discussion is out of scope at all.

Reply:
I had that question myself. But the issue is about trimming leading spaces: it is a legitimate problem affecting users that need to be solved.

If there is an inconsistency between the behaviour on leading spaces and non-leading spaces, it may (or may not?) be a problem, but it is irrelevant to what is being addressed right now. You need to know the difference.

Reply:
I agree the issue needs to be addressed quickly but I can't foresee when a discussion drags on.

If we only focus on fixing the issue itself, then are we going to come back to revisit the discussion? If we are I think that's fine but the question is whether we will once we fix the issue.

This bug probably stemmed from the incomplete handling of whitespaces and may possibly be the best approach at that time among the priorities.

Similarly, it may not be as important yet again depending on how the discussion goes but at the time I raised this I thought a simple replacement with `trimAndCollapseFrom` is sufficient.

Reply:
> are we going to come back to revisit the discussion

If you feel so strongly about it, then you take the action. No need to wait for others, right?

Reply:
Well, I can't do the thing where once say this PR is merged and go on to propose immediately to use `trimAndCollapseFrom` when @whipermr5 may have good reasons not to.

I could open an issue, but I can't ensure the discussion continues to take place for various reasons. With that said, I think I made my point so I will create a corresponding issue to continue the discussion.

Reply:
I decided against opening a new issue because we don't have the resources to take this further :P.